### ISSUE TITLE
TH1 Integral incorrectly returns nan

### ISSUE THREAD

**Issue by Dominic-Stafford** [root-project/root#6427]
Dear ROOT experts

I have produced some neural network output score, which I exported to ROOT histograms using uproot (as I am more familiar with that than pyROOT), so that I could set limits using the Higgs Combine tool. However in the Higgs Combine tool I encountered errors due to these histograms having nan integral. I presumed this problem may just have been caused by me outputting the histogram incorrectly, however when I investigated one of these histograms I found it had the property that hist.Integral(100, 100) gives nan, but hist.GetBinContent(100) gives 0.0, when I would assume these should always give the same result. What is the difference between these, please, and can anyone advise on where I might be going wrong?

Thank you

Dominic Stafford

**Comment by lmoneta** [comment]
Hi, 
It is strange that TH1::Integral(100,100) returns a different result than  GetBinContent(100). I would need you to send me a root file containing the histogram 

Lorenzo

**Comment by Dominic-Stafford** [comment]
Dear Lorenzo

I tried to produce an example ROOT file containing a problematic histogram, however when I checked the histogram that caused the crash in the Combine program by reading it directly from the file, the Integral() method worked without a problem. I then assumed Combine had modified the histogram in some way that caused this behaviour, so I attempted using pdb to save the histogram directly after the the crash, however this histogram also seems well-behaved. I will comment again if I manage to get the issue in a slightly more reproducible form, as I appreciate this isn't possible to debug from your side.
